Entity-ngsiv2-schema.json

```diff
@@ -0,0 +1,185 @@
+    "EntityIdentifier": {
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 256,
+      "pattern": "^[\\w\\-\\.\\{\\}\\$\\+\\*\\[\\]`|~^@!,:\\\\]+$"
+    },
+    "Name": {
+      "type": "string",
+      "pattern": "^(\\d|[a-zA-Z]|_)+$",
+      "minLength": 1,
+      "description": "NGSIv2 Name"
+    },
+    "Value": {
+      "type": [
+        "number",
+        "boolean",
+        "array",
+        "object",
+        "null",
+        "string"
+      ]
+    },
+    "TypeValue": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/Name"
+            },
+            {
+              "not": {
+                "const": "DateTime"
+              }
+            }
+          ]
+        },
+        "value": {
+          "$ref": "#/definitions/Value"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "DateTimeValue": {
+      "type": "string",
+      "format": "date-time"
+    },
+    "DateTimeTypeValue": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "const": "DateTime"
+        },
+        "value": {
+          "$ref": "#/definitions/DateTimeValue"
+        }
+      },
+      "required": [
+        "type",
+        "value"
+      ]
+    },
+    "GeoTypeValue": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "const": "geo:json"
+        },
+        "value": {
+          "$ref": "http://json-schema.org/geojson/geometry.json#"
+        }
+      },
+      "required": [
+        "type",
+        "value"
+      ]
+    },
+    "Metadata": {
+      "type": "object",
+      "additionalProperties": {
+        "anyOf": [
+          {
+            "$ref": "#/definitions/TypeValue"
+          },
+          {
+            "$ref": "#/definitions/DateTimeTypeValue"
+          }
+        ]
+      }
+    },
+    "dateCreated": {
+      "$ref": "#/definitions/DateTimeTypeValue"
+    },
+    "dateModified": {
+      "$ref": "#/definitions/DateTimeTypeValue"
+    },
+    "Attribute": {
+      "allOf": [
+        {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/DateTimeTypeValue"
+            },
+            {
+              "$ref": "#/definitions/GeoTypeValue"
+            },
+            {
+              "$ref": "#/definitions/TypeValue"
+            }
+          ]
+        },
+        {
+          "type": "object",
+          "properties": {
+            "metadata": {
+              "$ref": "#/definitions/Metadata"
+            }
+          }
+        }
+      ]
+    },
+    "EntityFragment": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "$ref": "#/definitions/EntityIdentifier"
+        },
+        "type": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/Name"
+            },
+            {
+              "default": "Thing"
```

Comment:
We can also take into account the default for attribute types, as described in "Partial Representations". Would it be possible?

Reply:
It would complicate the schema unnecessarily 

NTC